core/src/main/java/io/questdb/griffin/SqlParser.java

```diff
@@ -103,7 +104,24 @@ public class SqlParser {
         this.traversalAlgo = traversalAlgo;
         this.characterStore = characterStore;
         this.optimiser = optimiser;
-        this.expressionParser = new ExpressionParser(expressionNodePool, this, characterStore);
+        boolean sqlOperatorPrecedenceCompatMode = configuration.getSqlOperatorPrecedenceCompatMode();
+        if (!sqlOperatorPrecedenceCompatMode) {
+            this.expressionParser = new ExpressionParser(
```

Comment:
By default, we should be using the new registry as the active one and the legacy registry as the shadow one, but I don't see this here and in the below branch. Am I missing something?